lib/Expr/Parser.cpp

```diff
@@ -998,8 +998,8 @@ ExprResult ParserImpl::ParseParenExpr(TypeResult FIXME_UNUSED) {
   case 2:
     return ParseBinaryParenExpr(Name, ExprKind, IsFixed, ResTy);
   case 3:
-    if (ExprKind == Expr::Select)
-      return ParseSelectParenExpr(Name, ResTy);
+    assert(ExprKind == Expr::Select);
+    return ParseSelectParenExpr(Name, ResTy);
```

Comment:
I don't think, this is the right solution, because it makes it more diffcult to add more 3 argument expressions. I think it would be better to just do
```
...
else
  assert(0 && "Unknown tenrary expressions");
  return ExprResult();
```

That way if people are adding more expression they don't have to get rig of the assert.

Reply:
While I much prefer to have readable code now than to prepare for an eventuality that will probably never come, I can rewrite that change if you really want it.

Reply:
It might not come for mainline KLEE, but adding Expr seems like a common thing to want to do in an extension, so it makes it harder for them. I guess it's a really minor point, don't rewrite just for me.

Reply:
The updated version addresses your wishes.